src/WalkingModule/src/Module.cpp

```diff
@@ -922,23 +941,80 @@ bool WalkingModule::updateModule()
+
+            // DCM
+            data.vectors["dcm::position::measured"].assign(m_FKSolver->getDCM().data(), m_FKSolver->getDCM().data() + m_FKSolver->getDCM().size());
+            data.vectors["dcm::position::desired"].assign(m_DCMPositionDesired.front().data(), m_DCMPositionDesired.front().data() + m_DCMPositionDesired.front().size());
+            data.vectors["dcm::velocity::desired"].assign(m_DCMVelocityDesired.front().data(), m_DCMVelocityDesired.front().data() + m_DCMVelocityDesired.front().size());
+
+            // ZMP
+            data.vectors["zmp::measured"].assign(measuredZMP.data(), measuredZMP.data() + measuredZMP.size());
+            data.vectors["zmp::desired"].assign(desiredZMP.data(), desiredZMP.data() + desiredZMP.size());
+            // "zmp_des_planner_x", "zmp_des_planner_y",
+            iDynTree::Vector2& desiredZMPPlanner = m_desiredZMP.front();
+            data.vectors["zmp::desired_planner"].assign(desiredZMPPlanner.data(), desiredZMPPlanner.data() + desiredZMPPlanner.size());
+
+            // COM
+            data.vectors["com::position::measured"].assign(measuredCoM.data(), measuredCoM.data() + measuredCoM.size());
+
+            // Manual definition of this value to add also the planned CoM height
+            std::vector<double> CoMPositionDesired(3);
+            CoMPositionDesired[0] = m_stableDCMModel->getCoMPosition().data()[0];
+            CoMPositionDesired[1] = m_stableDCMModel->getCoMPosition().data()[1];
+            CoMPositionDesired[2] = m_retargetingClient->comHeight() + m_comHeightOffset;
+            data.vectors["com::position::desired"].assign(CoMPositionDesired.begin(), CoMPositionDesired.begin() + CoMPositionDesired.size());
+            data.vectors["com::position::desired_macumba"].assign(desiredCoMPosition.data(), desiredCoMPosition.data() + desiredCoMPosition.size());
+
+            // Manual definition of this value to add also the planned CoM height velocity
+            std::vector<double> CoMVelocityDesired(3);
+            CoMVelocityDesired[0] = m_stableDCMModel->getCoMVelocity().data()[0];
+            CoMVelocityDesired[1] = m_stableDCMModel->getCoMVelocity().data()[1];
+            CoMVelocityDesired[2] = m_retargetingClient->comHeightVelocity();
+            data.vectors["com::velocity::desired"].assign(CoMVelocityDesired.begin(), CoMVelocityDesired.begin() + CoMVelocityDesired.size());
+
+            // Left foot
+            data.vectors["left_foot::position::measured"].assign(leftFoot.getPosition().data(), leftFoot.getPosition().data() + leftFoot.getPosition().size());
+            data.vectors["left_foot::position::desired"].assign(m_leftTrajectory.front().getPosition().data(), m_leftTrajectory.front().getPosition().data() + m_leftTrajectory.front().getPosition().size());
+            data.vectors["left_foot::orientation::measured"].assign(leftFoot.getRotation().asRPY().data(), leftFoot.getRotation().asRPY().data() + leftFoot.getRotation().asRPY().size());
```

Comment:
The function `asRPY()` returns a copy of the object. It cannot be used in this way as it produces unexpected results. Indeed, I checked the logged orientation and it contains 7 values instead of 3 as instead expected.

@S-Dafarra @paolo-viceconte

Reply:
Thanks @isorrentino! Fixed in https://github.com/robotology/walking-controllers/pull/97/commits/45a8c20cfdb00e85d78d31137deb9d1a87365d65. The logged feet orientations, both measured and desired, are now 3 as expected. From a few tests in simulation, they seem to have reasonable values.